NorwegianRefugeeCouncil/core
# [CORE24-274] feat(households): auto create household when creating individual

Jira ticket: JIRA-CORE24-274


## What have you implemented?

Added the link between households and individuals:
- When creating a new individual -> A household is created first, and then the individual belonging to that household, marking the individual as head of household. All in a single database transaction.
- When creating a household -> all the individuals in its definition are also created and linked to the household in a single db transaction.

Note that db transaction now can be handled in the service layer, to allow for more complex cases affecting multiple entities.
They are still supported at the data access layer, sent as an optional parameter. We may want to revisit this later on and make it mandatory, so we don't need to handle them at that layer too.

## How to test it?

Create an individual, expect householdId in the response.
Create a household with individuals, expect them in the response. Then request the individuals and they should have the householdId too.

## Extra/next steps
